crates/nu-command/tests/commands/unfold.rs

```diff
@@ -0,0 +1,64 @@
+use nu_test_support::{nu, pipeline};
+
+#[test]
+fn unfold_single_element_break() {
+    let actual = nu!("unfold 1 {|x| if $x == 3 { [$x] } else { [$x, ($x + 1)] }} | to nuon");
+
+    assert_eq!(actual.out, "[1, 2, 3]");
+}
+
+#[test]
+fn unfold_single_value_break() {
+    let actual = nu!("unfold 1 {|x| if $x == 3 { $x } else { [$x, ($x + 1)] }} | to nuon");
+
+    assert_eq!(actual.out, "[1, 2, 3]");
+}
+
+#[test]
+fn unfold_nothing_break() {
+    let actual = nu!("unfold 1 {|x| if $x <= 3 { [$x, ($x + 1)] }} | to nuon");
+
+    assert_eq!(actual.out, "[1, 2, 3]");
+}
+
+#[test]
+fn unfold_invalid_return() {
+    let actual = nu!("unfold 0 {|x| [a b c]}");
+    assert!(actual.err.contains("Invalid block return"));
+}
+
+#[test]
+fn unfold_allows_nothing_state() {
+    let actual = nu!(pipeline(
+        r#"
+        unfold 0 {|x|
+          if $x == $nothing {
```

Comment:
because of https://github.com/nushell/nushell/pull/10478, could you replace all `$nothing`s with `null`? :yum:

Reply:
Fixed

Reply:
:pray: